### PR TITLE
merge from sujap

### PR DESCRIPTION
**Pull request by appu226**

### CODE REVIEW

File: algo_linear/algoLinear.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+#include <vector>
+#include <stdexcept>
+#include <string>
+#include <algorithm>
+
+namespace space {
+
+	
+
+	AlgoLinearDepthOne::AlgoLinearDepthOne(std::vector<double> wtsVec)
```
Comment (appu226) [inline algo_linear/algoLinear.cpp:12]: why not take const reference to vector?

File: algo_linear/algoLinear.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+
+
+// Depth Two algos
+
+
+	AlgoLinearDepthTwoExt::AlgoLinearDepthTwoExt(int _breadth, std::vector<double> wtsVec)
+	{
+		space_assert(breadth > 0, "Breadth must be positive");
+		space_assert(wtsVec.size() == 5, "Need 6 weights");
```
Comment (appu226) [inline algo_linear/algoLinear.cpp:60]: text should say "Need 5 weights"

File: algo_linear/algoLinear.cpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+
+
+// Depth Two algos
+
+
+	AlgoLinearDepthTwoExt::AlgoLinearDepthTwoExt(int _breadth, std::vector<double> wtsVec)
+	{
+		space_assert(breadth > 0, "Breadth must be positive");
```
Comment (appu226) [inline algo_linear/algoLinear.cpp:59]: Do you want to check _breadth instead of breadth?

File: algo_linear/algoLinear.cpp [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+		std::vector<ScoreTriple> allScores = this->getAllScores(board);
+		int direction = board->whoPlaysNext() == Color::White ? 1 : -1;
+
+		auto cmp = [direction](const ScoreTriple& a, const ScoreTriple& b) {
+			return direction * (std::get<2>(a) < std::get<2>(b)) < 0;
+		};
+
+
+		std::sort(allScores.begin(), allScores.end(), cmp);
+		allScores.resize(this->breadth);
```
Comment (appu226) [inline algo_linear/algoLinear.cpp:135]: what if size is already  less than breadth?

File: chess/board_impl.cpp [hunk 5]
```diff
@@ -34,32 +36,88 @@ namespace {
...
 	{
 		if (m_pieces[position.rank][position.file].pieceType == PieceType::None)
 			return std::optional<Piece>();
 		else
 			return m_pieces[position.rank][position.file];
 	}
+
+	// Castling conditions:
+	// King & Rook not moved -- examined here using booleans, updated in updateBoard
+	// No piece in between -- examined here directly
```
Comment (sujap) [inline chess/board_impl.cpp:108]: checked elsewhere

**Pull request merged**